overlord/devicestate/devicectx.go

```diff
@@ -104,6 +105,27 @@ func (dc groundDeviceContext) HasModeenv() bool {
 	return dc.model.Grade() != asserts.ModelGradeUnset
 }
 
+// IsCoreLegacy is true for UC16/18
+func (d *groundDeviceContext) IsCoreLegacy() bool {
+	return !d.HasModeenv() && !release.OnClassic
```

Comment:
now that I think more, this is strange, why use release.OnClassic vs d.model.Classic() ?

Reply:
Changed